native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -236,4 +239,29 @@ public static BString getInterceptorName(BObject interceptor) {
         ServiceType serviceType = (ServiceType) interceptor.getType();
         return StringUtils.fromString(serviceType.getName());
     }
+
+    public static Object getResourceAnnotation(BObject service, BArray path, BString methodName) {
+        BString identifier = StringUtils.fromString(getModule().toString() + COLON + RESOURCE_CONFIG);
+        ServiceType serviceType = (ServiceType) service.getType();
+        ResourceMethodType resourceMethod = (ResourceMethodType) getResourceMethod(serviceType, getPathList(path));
+        if (resourceMethod != null) {
+            return resourceMethod.getAnnotation(identifier);
+        }
+        RemoteMethodType remoteMethod = getRemoteMethod(serviceType, String.valueOf(methodName));
+        if (remoteMethod != null) {
+            return remoteMethod.getAnnotation(identifier);
+        }
+        return null;
+    }
+
+    private static Object getResourceMethod(ServiceType serviceType, List<String> pathList) {
+        for (ResourceMethodType resourceMethod : serviceType.getResourceMethods()) {
+            String accessor = resourceMethod.getAccessor();
+            if ((GET_ACCESSOR.equals(accessor) || SUBSCRIBE_ACCESSOR.equals(accessor)) &&
```

Comment:
We don't need to check the accessors here, right? What happens if there are two resources with the same name, one for Query, and one for Subscription? How do we process this? What if only one of them has a field interceptor?

Reply:
Handled above scenarios in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1316/commits/6346096ae72baf454266869638611f2fb109e779